# Module 54 & 55: Yield and Yield Spread Measures for Fixed-Rate Bonds
<hr>

In [1]:
# Import libraries
import math
import pandas as pd
import numpy as np
import numpy_financial as npf
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from pandas.core.common import random_state
from sklearn.linear_model import LinearRegression

### Yield Measures for Fixed-Rate Bonds

The number of bond coupon payments per year is referred to as the periodicity of a bond. A bond with a **periodicity** of 2 will have its YTM quoted on a **semiannual bond basis**. For a given coupon rate, the greater the periodicity, the more compounding periods, and the greater the effective annual yield (which reflects compounding).

In general, the effective annual yield for a bond with its YTM stated for a periodicity of $n$ (n compounding periods per year) is as follows:

$\qquad\text{annual yield} = \big( 1 + \frac{\text{YTM}}{n} \big)^n − 1$

**Example: Effective annual yields**

What is the effective annual yield for a bond with a stated YTM of 10%:

1. When the periodicity of the bond is 2 (pays semiannually)?
2. When the periodicity of the bond is 4 (pays quarterly)?


**Answer:**

1. $\qquad\text{annual yield} = \big(1 + \frac{0.10}{2}\big)^2 – 1 = 1.05^2 – 1 = 0.1025 = 10.25\%$
2. $\qquad\text{annual yield} = \big(1 + \frac{0.10}{4}\big)^4 – 1 = 1.025^4 – 1 = 0.1038 = 10.38\%$

In [2]:
## Example:  Calculat Effective annual yield for a bond with a stated YTM of 10%


# Example 1: When the periodicity of the bond is 2 (pays semiannually)
annual_yield1 = 1.05**2 - 1
print("Effective annual yield of semiannual 10% YTM bond example = %",(float(annual_yield1)*100))


# Example 2: When the periodicity of the bond is 4 (pays quarterly)
annual_yield2 = 1.025**4 - 1
print("Effective annual yield of quarterly  10% YTM bond example = %",(float(annual_yield2)*100))

Effective annual yield of semiannual 10% YTM bond example = % 10.250000000000004
Effective annual yield of quarterly  10% YTM bond example = % 10.381289062499954


**Example: Adjusting yields for periodicity**

An Atlas Corporation bond is quoted with a YTM of 4% on a semiannual bond basis. What yields should be used to compare it with a quarterly-pay bond and an annual-pay bond?

To compare this with the yield on an annual-pay bond, which is an effective annual yield, we need to calculate the effective annual yield on the semiannual coupon bond, which is $1.02^2 − 1 = 4.04\%$.

For the quoted annual YTM on the quarterly-pay bond, we need to calculate the effective quarterly yield and multiply by four. The quarterly yield (yield per quarter) that is equivalent to a yield of 2% per six months is $1.02^{1/2} − 1 = 0.995\%$. The quoted annual rate for the equivalent yield on a quarterly bond basis is `4 × 0.995% = 3.98%`.

Note that we have shown that the effective annual yields are the same for the following:

* An annual coupon bond with a yield of `4.04%` on an annual basis (periodicity of one)
* A semiannual coupon bond with a yield of `4.0%` on a semiannual basis (periodicity of two)
* A quarterly coupon bond with a yield of `3.98%` on quarterly basis (periodicity of four)

In [3]:
## Adjust Yields for Periodicity
semi_4    = 1.02**2 - 1
quarter_4 = 1.02**(0.5) -1 

print("YTM of 4% on an annual bond basis      = % 4.00")
print("YTM of 4% on a  semi-annual bond basis = %", (round(semi_4,5)*100))
print("YTM of 4% on a  quarterly bond basis   = %", (round(quarter_4,5)*100*4))

YTM of 4% on an annual bond basis      = % 4.00
YTM of 4% on a  semi-annual bond basis = % 4.04
YTM of 4% on a  quarterly bond basis   = % 3.9800000000000004


Bond yields calculated using the stated coupon payment dates are referred to as following the street convention. When coupon dates fall on weekends and holidays, coupon payments will actually be made the next business day. The yield calculated using these actual coupon payment dates is referred to as the true yield. Because coupon payments will be made later when holidays and weekends are taken into account, true yields are usually slightly lower than street convention yields, if only by a few basis points.

**Current yield** (also called **income yield** or **running yield**) looks at just one source of return, which is *a bond's annual interest income—* it does not consider capital gains or losses or reinvestment income. The formula for the current yield is as follows:


$\qquad\large\text{current yield} = \frac{\text{annual cash coupon payment}}{\text{bond price}} $


**Example: Computing current yield**

Consider a 20-year, \\$1,000 par value, 6\% semiannual-pay bond that is currently trading at a flat price of \\$802.07. Calculate the current yield.

**Answer:**

The annual cash coupon payments total:

   annual cash coupon payment = par value × stated coupon rate = \\$1,000 × 0.06 = \\$60

Because the bond is trading at \\$802.07, the current yield is:

$\qquad\large\text{current yield} = \large\frac{60}{802.07} = 0.0748 \text{ or 7.48%}$

Note that current yield is based on annual coupon interest so that it is the same for a semiannual-pay and annual-pay bond with the same coupon rate and price.


In [4]:
## Consider a 20-year, $1,000 par value, 6% semiannual-pay bond 
## that is currently trading at a flat price of $802.07. Calculate the current yield.


# Calculate Annual Coupon
annual_coupon1 = 1000 * 0.06


# Calculate Current Yield
current_yield1 = annual_coupon1 / 802.07


#Print output
print("Consider a 20-year, $1,000 par value, 6% semiannual-pay bond \n that is currently trading at a flat price of $802.07. \n  Calculate the current yield.")
print("_______________________________________")
print("")
print("Annual Coupon of the Bond is = $",annual_coupon1)
print("_______________________________________")
print("")
print("Current yield of the Bond is = %",round(current_yield1,4))
print("Current yield of the Bond is = %",round(current_yield1,4)*100)

Consider a 20-year, $1,000 par value, 6% semiannual-pay bond 
 that is currently trading at a flat price of $802.07. 
  Calculate the current yield.
_______________________________________

Annual Coupon of the Bond is = $ 60.0
_______________________________________

Current yield of the Bond is = % 0.0748
Current yield of the Bond is = % 7.48


A bond's **simple yield** takes a discount or premium into account by assuming that any discount or premium declines evenly over the remaining years to maturity. The sum of the annual coupon payment plus (minus) the straight-line amortization of a discount (premium) is divided by the flat price to get the simple yield.

**Example: Computing simple yield**

A 3-year, 8\% coupon, semiannual-pay bond is priced at 90.165. Calculate the simple yield.

**Answer:**

The discount from par value is `100 – 90.165 = 9.835`. 
Annual straight-line amortization of the discount is `9.835 / 3 = 3.278`:

$\qquad\large\text{simple yield} = \large\frac{8 + 3.278}{90.165} = 0.0125 \text{ or 12.5%} $



In [5]:
## A 3-year, 8% coupon, semiannual-pay bond is priced at 90.165. Calculate the simple yield.


## Calculate Simple Yield
simple_yield1 = (8 + 3.278) / 90.165


##Print output
print("A 3-year, 8% coupon, semiannual-pay bond is priced at 90.165. Calculate the simple yield.")
print("_______________________________________")
print("")
print("Simple yield of the Bond is = %",round(simple_yield1,4))
print("Simple yield of the Bond is = %",round(simple_yield1,3)*100)

A 3-year, 8% coupon, semiannual-pay bond is priced at 90.165. Calculate the simple yield.
_______________________________________

Simple yield of the Bond is = % 0.1251
Simple yield of the Bond is = % 12.5


<hr>

**Note**: *Done fastest on Texas Instruments BAII Calculator using TVOM functions*


### Callable Bonds
For a **callable bond**, an investor's yield will depend on whether and when the bond is called. The **yield to call** can be calculated for each possible call date and price. The lowest of YTM and the various yields to call is termed the **yield to worst**. The following example illustrates these calculations.

**Example: Yield to call and yield to worst**

Consider a 5-year, semiannual-pay 6% bond trading at 102 on January 1, 20X4. 
The bond is callable according to the following schedule:

* Callable at 102 on or after January 1, 20X7
* Callable at 101 on or after January 1, 20X8

Calculate the bond's YTM, yield to first call, yield to second call, and yield to worst.

**Answer**

The yield to maturity on the bond is calculated as:

    N = 10; PMT = 3; FV = 100; PV = –102; CPT → I/Y = 2.768%

    2 × 2.768 = 5.54% = YTM

To calculate the yield to first call, we calculate the YTM using the number of semiannual periods until the first call date in 20X7 (6) for N and the call price (102) for FV:

    N = 6; PMT = 3; FV = 102; PV = –102; CPT → I/Y = 2.941%

    2 × 2.941 = 5.88% = yield to first call

To calculate the yield to second call, we calculate the YTM using the number of semiannual periods until the second call date in 20X8 (8) for N and the call price (101) for FV:

    N = 8; PMT = 3; FV = 101; PV = –102; CPT → I/Y = 2.830%

    2 × 2.830 = 5.66% = yield to second call

The lowest yield, 5.54%, is realized if the bond is held to maturity and not called, so the *yield to worst* is 5.54%.

$\qquad\text{Yield to worst} = 5.54\%$

A callable bond can be viewed as an equivalent straight (option-free) bond combined with a short call option position (because the right to call the bond lies with the issuer, not the investor):

    callable bond value = straight bond value – call option value


Stated differently, we can view the value of an equivalent straight bond (known as the **option-adjusted price**) as the value of the callable bond plus the value of the call option embedded in the callable bond (which could be derived from an option pricing model).

This option-adjusted price can then be used to calculate an **option-adjusted yield**, which represents the yield that the bond would be offering if it were not callable. Because the existence of the call option decreases the bond price and increases an investor's required yield, "removing" the option in this manner will cause the option-adjusted yield to be *lower* than the yield of the callable bond. The option-adjusted yield is useful because it can be used to compare the yields of bonds with various embedded options to each other and to similar option-free bonds on a consistent basis.


**Professor's Note**

Take care to understand that option-adjusted prices and yields remove the impact of the option from a bond with an embedded option. It is a common mistake when first encountering these measures to incorrectly think they are adjusting to incorporate the impact of the option, when in fact they are doing the opposite. When you see `"option-adjusted,"` think `"option removed"` or `"option taken away."`

**Matrix pricing** is a method of estimating the required YTM (or price) of bonds that are currently not traded, or infrequently traded. The procedure is to use the YTMs of traded bonds that have credit quality very close to that of a nontraded or infrequently traded bond and are similar in maturity and coupon, to estimate the required YTM.

<br>

**EXAMPLE: PRICING AN ILLIQUID BOND**

Rob Phelps, CFA, is estimating the value of a nontraded 4% annual-pay, A+ rated bond that has three years remaining until maturity. He has obtained the following yields to maturity on similar corporate bonds:

* A+ rated, 2-year annual-pay, YTM = 4.3%
* A+ rated, 5-year annual-pay, YTM = 5.1%
* A+ rated, 5-year annual-pay, YTM = 5.3%

Estimate the value of the nontraded bond.

**Answer:**

* **Step 1**:	Take the average YTM of the 5-year bonds:   

    (5.1 + 5.3) / 2 = 5.2%

--------------------
* **Step 2**:	Interpolate the 3-year YTM based on the 2-year and average 5-year YTMs:

    4.3% + (5.2% − 4.3%) × [(3 years − 2 years) / (5 years − 2 years)] = 4.6%


--------------------

* **Step 3**:	Price the nontraded bond with a YTM of 4.6%:

    N = 3; PMT = 4; FV = 100; I/Y = 4.6; CPT → PV = -98.354



The estimated value is \\$98.354 per \\$100 par value.

--------------------

In **Step 2** in the preceding example, we have used `simple linear interpolation`. Because the maturity of the nontraded bond is three years, we estimate the YTM on the 3-year bond as the yield on the 2-year bond, plus `one-third` of the difference between the YTM of the 2-year bond and the average YTM of the 5-year bonds. (The difference between the maturities of the 2-year bond and the 3-year bond is one year, and the difference between the maturities of the 2-year and 5-year bonds is three years.)

A variation of matrix pricing used for new bond issues focuses on spreads. The required yield spread to a benchmark for a new issue bond can be estimated by observing spreads on existing similar securities, as demonstrated in the following example.

--------------------
**EXAMPLE: ESTIMATING THE SPREAD FOR A NEW 6-YEAR, A RATED BOND ISSUE**

Consider the following market yields:

* 4-year, U.S. Treasury bond, YTM 1.48%
* 5-year, A rated corporate bond, YTM 2.64%
* 6-year, U.S. Treasury bond, YTM 2.15%

Estimate the required yield spread on a newly issued 6-year, A rated corporate bond.



In [6]:
## ESTIMATING THE SPREAD FOR A NEW 6-YEAR, A RATED BOND ISSUE

## The known YTM on the 3 listed bonds
yr4_Treasury = .0148 * 100
yr5_A_rated  = .0264 * 100
yr6_Treasury = .0215 * 100

## Use interpolation on 4-yr & 6-yr Treasury to find interpolated 5-year Treasury YTM
yr5_Interpol_T = yr4_Treasury + ((yr6_Treasury - yr4_Treasury) * (1/2))


## Calculate the yield spread on existing 5-year corporate debt 
yr5_C_spread = yr5_A_rated - yr5_Interpol_T

## We will apply this yield spread to the new 6-year corporate debt issue
yr6_Interpol_C = yr6_Treasury + yr5_C_spread


## Print Output of the Original & interpolated Bonds
print("4-year, U.S. Treasury bond, YTM                    = %", yr4_Treasury)
print("5-year, A rated corporate bond, YTM                = %", yr5_A_rated)
print("6-year, U.S. Treasury bond, YTM                    = %", yr6_Treasury)
print("--------------------------------------------------")
print("The yield spread on existing 5-year corporate debt = %",round(yr5_C_spread,4))
print("--------------------------------------------------")
print("5-year, U.S. Treasury Bond, YTM Interpolated       = %",round(yr5_Interpol_T,4))
print("6-year, A rated corporate bond, YTM Interpolated   = %",round(yr6_Interpol_C,4))


4-year, U.S. Treasury bond, YTM                    = % 1.48
5-year, A rated corporate bond, YTM                = % 2.64
6-year, U.S. Treasury bond, YTM                    = % 2.15
--------------------------------------------------
The yield spread on existing 5-year corporate debt = % 0.825
--------------------------------------------------
5-year, U.S. Treasury Bond, YTM Interpolated       = % 1.815
6-year, A rated corporate bond, YTM Interpolated   = % 2.975


**Answer:**

We will use the existing 5-year, A rated corporate bond to estimate the required yield spread of the issuer by comparing the YTM of the 5-year corporate bond to the interpolated 5-year Treasury bond YTM.

Interpolated 5-year Treasury bond YTM:

    = 1.48% + (2.15% − 1.48%) × [(5 years − 4 years) / (6 years − 4 years)]

    = 1.815%

**Note**: Because the target maturity of the existing bond (5 years) is midway between the two Treasury bond maturities (4 and 6 years), we could have simply averaged the two Treasury bond yields here
(1.48% + 2.15%) / 2 = 1.815%.

The yield spread on existing 5-year corporate debt is 2.64% – 1.815% = 0.825%.

We will apply this yield spread to the new 6-year corporate debt issue:

YTM for the new 6-year corporate bond = 2.15% + 0.825% = 2.975%

<hr>

### LOS 55.b: Compare, calculate, and interpret yield and yield spread measures for fixed-rate bonds.

A **yield spread**, or **benchmark spread**, is the difference between the yields of a bond and a benchmark security. For example, if a `5-year corporate bond has a yield of 6.25%` and its benchmark, `the 5-year Treasury note, has a yield of 3.50%`, the corporate bond has a benchmark spread of `625 − 350 = 275 basis points`.

For fixed-coupon bonds, on-the-run government bond yields for the same or nearest maturity are frequently used as benchmarks because they are the most actively traded bonds, and therefore give the most useful price and yield information. A yield spread in basis points over a government bond is also known as a **G-spread**. If a benchmark government bond with exactly the same maturity as the riskier bond does not exist, interpolation should be used to estimate the appropriate maturity benchmark yield.

<hr>

**Example: G-Spread**

A 3-year, 8% coupon, semiannual-pay bond is priced at 103.165, and 1-year and 4-year U.S. Treasury yields are 3% and 5%, respectively. Calculate the G-spread of the bond.

In [7]:
## EXAMPLE: CALCULATE G-SPREAD

## Define our 1-year and 4-year US treasury Yields
UST_1yr = 3.00
UST_4yr = 5.00


## Find the yield to maturity of the 3-year, 8% coupon, semiannual-pay bond priced at 103.165
## Using numpy_financial function to find the YTM convert decimals and Quote it by multiplying by 2
GBondYTM = npf.rate(nper=6, pmt=4, pv=-103.165, fv=100) * 100 * 2


## Interpolate the quoted YTM of the 3-year government bond 
GBond_Interpol =  UST_1yr + ((3 - 1)/(4 - 1)) * (UST_4yr - UST_1yr)


## Calculate the G-Spread of the interpolated bonds
G_Spread = GBondYTM - GBond_Interpol


## PRINT RESULTS
print("3-year, 8% Coupon Semiannual U.S. Treasury bond,  YTM = %", round(GBondYTM,2))
print("3-year, Government Bond with Interpolated yield,  YTM = %", round(GBond_Interpol,2))
print("")
print("The G-spread of the Interpolated US T-bond yields     = %", round(G_Spread,2))
print("")
print("The basis point spread of the G-spread is 248 basis points")

3-year, 8% Coupon Semiannual U.S. Treasury bond,  YTM = % 6.82
3-year, Government Bond with Interpolated yield,  YTM = % 4.33

The G-spread of the Interpolated US T-bond yields     = % 2.48

The basis point spread of the G-spread is 248 basis points


**Answer:**

The YTM of the bond is calculated as:

    N = 6; PMT = 4; FV = 100; PV = –103.165; CPT → I/Y = 3.408%

    Quoted YTM = 2 × 3.408% = 6.82%

The interpolated 3-year government bond yield is:

    3% + [(3 – 1) / (4 – 1)] × (5% – 3%) = 4.33%

Then, the `G-spread = 6.82% – 4.33% = 2.48%`, or 248 basis points.

<hr>

An alternative to using government bond yields as benchmarks is to use rates for interest rate swaps in the same currency and with the same tenor as a bond. Yield spreads relative to swap rates are known as **interpolated spreads** or **I-spreads** and represent the extra return of a bond in excess of the [interbank market reference rates](https://www.ice.com/iba/ice-swap-rate) (MRRs) used in swap contracts. I-spreads are frequently stated for bonds denominated in <u>euros</u>.

Yield spreads are useful for analyzing the factors that affect a bond's yield. If a corporate bond's yield increases from 6.25% to 6.50%, this may have been caused by factors that affect all bond yields (macroeconomic factors) or by firm-specific or industry-specific (microeconomic) factors. If a bond's yield increases but its spread remains the same, the yield on its benchmark must have also increased, which suggests macroeconomic factors caused bond yields in general to increase. However, if the yield spread increases, this suggests the increase in the bond's yield was caused by microeconomic factors, such as credit risk of the issuer increasing or the issue's liquidity deteriorating.

**Professor's Note**

Recall from Quantitative Methods that an interest rate is composed of the real risk-free rate, the expected inflation rate, and a risk premium. We can think of macroeconomic factors as those that affect the real risk-free rate and expected inflation (which make up the benchmark yield), and microeconomic factors as those that affect credit and liquidity risk premiums (which make up the yield spread). Differences in taxation of the returns from bonds of different issuers can also affect yield spreads.


**Links of Interest**
1. [What is Yield Z/G-spread?](https://fbs.com/glossary/yield-spread-68)
2. [ICE Swap Rate® (formerly known as ISDAFIX)](https://www.ice.com/iba/ice-swap-rate)
3. [St Louis Fed: ICE BofA US High Yield Index Option-Adjusted Spread](https://fred.stlouisfed.org/series/BAMLH0A0HYM2)
4. [Interpolated Yield Curve (I Curve): Definition and Uses](https://www.investopedia.com/terms/i/interpolated_yield_curve.asp)

#### Zero-Volatility and Option-Adjusted Spreads

The G-spread and I-spread are based on the difference between the yields of a specific bond and a benchmark. Recall that the YTM of a bond is the single discount rate that sets the present value of the cash flows of the bond equal to its market price.

We can observe from the prices of zero-coupon bonds that different individual cash flows occurring at different maturities can have different yields. Yields earned by individual cash flows at different maturities are referred to as **spot rates**. The single YTM of a coupon-paying bond represents a weighted average of the different spot rates offered by the individual cash flows of the bond.

**Professor's Note**

Spot rates are formally introduced and discussed later in our reading on The Term Structure of Interest Rates. For now, it is enough to know that calculating spreads over benchmark spot rates is a more precise way of calculating spreads than basing the measure on YTMs because it better captures how rates vary over different maturities (referred to as the term structure of rates).

A method for deriving a bond's yield spread to a benchmark spot yield curve that accounts for the shape of the yield curve is to add an equal amount to each benchmark spot rate and value the bond with those rates. When we find an amount which, when added to the benchmark spot rates, produces a value equal to the market price of the bond, we have the appropriate yield curve spread. A yield spread calculated this way is known as a zero-volatility spread or Z-spread.

<hr>

**Example: Zero-volatility spread**

The 1-, 2-, and 3-year spot rates on Treasuries are 4%, 8.167%, and 12.377%, respectively. 

Consider a 3-year, 9% annual coupon corporate bond trading at 89.464. 

The YTM is 13.50%, and the YTM of a 3-year Treasury is 12%. 

Calculate the G-spread and the Z-spread of the corporate bond.

In [8]:
## Calculate the G-spread from the example above

# Given Treasury Bond Yield-to-Maturities
YTM_Bond = 13.50
YTM_Treasury = 12.00

# G-Spread Calculation
G_Spread = YTM_Bond - YTM_Treasury
print("The G-Spread is = %", G_Spread)

The G-Spread is = % 1.5


**Answer:**

The G-spread is $\text{YTM}_\text{bond} − \text{YTM}_\text{Treasury} = 13.50 − 12.00 = 1.50\%.$

**Note** 

To compute the Z-spread, set the present value of the bond's cash flows equal to today's market price. Discount each cash flow at the appropriate zero-coupon bond spot rate plus a fixed-spread ZS. Solve for ZS in the following equation, and you have the Z-spread:

$89.464 = \Large\frac{9}{(1.04+ZS)^1} + \frac{9}{(1.08167+ZS)^2} + \frac{109}{(1.12377+ZS)^3}$


$\Rightarrow ZS = 1.67\%$, or 167 basis points

Note that this spread is found by trial and error. In other words, pick a number "ZS," plug it into the right-hand side of the equation, and see if the result equals 89.464. If the right-hand side equals the left, then you have found the Z-spread. If not, adjust ZS in the appropriate direction and recalculate.



<hr>

An [**option-adjusted spread (OAS)**](https://fred.stlouisfed.org/series/BAMLH0A0HYM2) is used for bonds with embedded options. Loosely speaking, the OAS takes the option yield component out of the Z-spread measure; the OAS is the spread to the government spot rate curve that the bond would have if it were option free. This is similar to the option-adjusted yields discussed earlier, the only difference being OAS respects the term structure of rates instead of being based on YTM.

If we calculate an OAS for a callable bond, it will be less than the bond's Z-spread. The difference is the extra yield required to compensate bondholders for the call option. That extra yield is referred to as the option value. Thus, we can write the following:

    option value = Z-spread − OAS

    OAS = Z-spread − option value

For example, if a callable bond has a Z-spread of 180 bp and the value of the call option is 60 bp, the bond's OAS is 180 – 60 = 120 bp.

The interpretation here is that investors are demanding the Z-spread (180 bp) for the credit, liquidity, taxation, and optionality risks of the callable bond. When the component of the spread relating to optionality is removed, we are left with the OAS (120 bp) that rewards investors for facing credit, liquidity, and taxation risks.

<hr>


**What is Yield Z/G-spread?**

Usually, debt instruments with different characteristics (maturity date/credit rating or risk) have different yields. To understand better what is spread, let’s take bond yields as an example and analyze the risks related to them. The bond yield is the return rate, which holders of bonds get if they have this bond until maturity and receive the cash flows at the promised dates. Risks include credit, interest rates, inflation, and others.

We can divide the measures of yield spread into the nominal spread (G-spread), interpolated spread (I-spread), zero-volatility spread (Z-spread), and option-adjusted spread (OAS).

**G-spread**
Nominal spread (G-spread) represents the difference between Treasury bond yields and corporate bond yields with the same maturity. Treasury bonds have zero default risk, which is why the difference between corporate and Treasury bonds shows the default risk. We can calculate the G-spread by using the following formula:

    G-Spread = corporate bond’s yield – government bond’s yield

**I-spread**
Interpolated spread (I-spread) is the difference between a bond's yield and the swap rate. We can use LIBOR as an example. It shows the difference between a bond's yield and a benchmark curve. If the I-spread increases, the credit risk also rises. I-spread is usually lower than the G-spread.

**Z-spread**
This type of spread is also known as a zero-volatility spread. Z-spread is added to each spot interest rate to make the present value of the bond's cash flows equal to the bond's price.

**Option-adjusted spread**
The option-adjusted spread is calculated as a zero-volatility spread minus the call option’s value. There is a term “spread” in the Forex market, too. It refers to the commission you pay a broker. The Forex spread is calculated as a difference between the bid and ask prices.

# Module Quiz 55.1 Examples
<hr>

<br>

1. Based on semiannual compounding, what is the YTM of a 15-year, zero-coupon, \\$1,000 par value bond that is currently trading at \\$331.40?

* **ANSWER** 7.500\%
* **EXPLANATION** 
    `N = 30; FV = 1,000; PMT = 0; PV = –331.40; CPT → I/Y = 3.750 × 2 = 7.500%`.
    
    
Alternatively, $\biggr[\big(\frac{1,000}{331.4}\big)^\frac{1}{30}−1 \biggr]\times 2 = 7.5 \%$.
(Module 55.1, LOS 55.a)

<br>


2. An analyst observes a Widget & Co. 7.125%, 4-year, semiannual-pay bond trading at 102.347. The bond is callable at 101 in two years. The bond's yield to call is closest to:

* **ANSWER** 6.3%
* **EXPLANATION** 
    `N = 4; FV = 101; PMT = 3.5625; PV = –102.347; CPT → I/Y = 3.167 × 2 = 6.334%`
    (Module 55.1, LOS 55.a)

<br>


3. Holding the effective annual yield constant, if the periodicity of a bond is increased, its stated YTM will:

* **ANSWER** decrease.
* **EXPLANATION** Due to their increased compounding frequency, bonds with higher periodicity will have a lower stated YTM for a certain level of effective annual yield (EAY). For example, for an EAY of 5% and periodicity of 2, the stated YTM must solve `[1 + (YTM / 2)]2 = 1.05`. Hence, `YTM = 2 (1.051/2 – 1) = 4.94%`. With periodicity of 4, the stated YTM must solve `[1 + (YTM / 4)]4 = 1.05`. Hence, `YTM = 4 (1.051/4 – 1) = 4.91%`. 
(Module 55.1, LOS 55.a)

<br>

4. A corporate bond is quoted at a spread of +235 basis points over an interpolated 12-year U.S. Treasury bond yield. This spread is a(n):


* **ANSWER** G-spread.
* **EXPLANATION** G-spreads are quoted relative to an actual or interpolated government bond yield. I-spreads are quoted relative to swap rates. Z-spreads are calculated based on the shape of the benchmark yield curve. 
(Module 55.1, LOS 55.b)

<br>

5. For a callable bond, relative to its option-adjusted spread, its Z-spread is most likely to be:


* **ANSWER** higher.
* **EXPLANATION** A callable bond will offer a higher yield than an equivalent straight bond because the investor faces the **call risk of the option**. Hence, the Z-spread, which includes the impact of the option, will be higher than the OAS, which has removed the impact of the option. 
(Module 55.1, LOS 55.b)



## Reading 55: Key Concepts

<hr>

### LOS 55.a

The effective yield of a bond depends on its periodicity, or frequency of coupon payments. For an annual-pay bond, the effective yield is equal to the yield to maturity (YTM). For bonds with greater periodicity, the effective yield is greater than the YTM.

A YTM quoted on a semiannual bond basis is two times the semiannual discount rate.

Bond yields that follow street convention use the stated coupon payment dates. A true yield accounts for coupon payments that are delayed by weekends or holidays and may be slightly lower than a street convention yield.

Current yield is the ratio of a bond's annual coupon payments to its price. Simple yield adjusts current yield by using straight-line amortization of any discount or premium.

For a callable bond, a yield to call may be calculated using each of its call dates and prices. The lowest of these yields or its YTM is a callable bond's yield to worst.

### LOS 55.b

A yield spread or benchmark spread is the difference between a bond's yield and a benchmark yield or yield curve. If the benchmark is a government bond yield, the spread is known as a government spread or G-spread. If the benchmark is a swap rate, the spread is known as an interpolated spread or I-spread.

A zero-volatility spread or Z-spread is the percentage spread that must be added to each spot rate on the benchmark yield curve to make the present value of a bond's cash flows equal to its price.

An option-adjusted spread (OAS) is used for bonds with embedded options and represents the spread the bond would offer if it had no embedded options. For a callable bond, the OAS is equal to the Z-spread minus the call option value in basis points.